massa-bootstrap/src/server.rs

```diff
@@ -139,26 +106,63 @@ struct BootstrapServer {
     keypair: KeyPair,
     bootstrap_config: BootstrapConfig,
     version: Version,
-    blacklist: Option<HashSet<IpAddr>>,
-    whitelist: Option<HashSet<IpAddr>>,
     ip_hist_map: HashMap<IpAddr, Instant>,
 }
 
+#[allow(clippy::result_large_err)]
+#[allow(clippy::type_complexity)]
+fn reload_whitelist_blacklist(
+    whitelist_path: &PathBuf,
+    blacklist_path: &PathBuf,
+) -> Result<(Option<HashSet<IpAddr>>, Option<HashSet<IpAddr>>), BootstrapError> {
+    let whitelist = if let Ok(whitelist) = std::fs::read_to_string(whitelist_path) {
+        Some(
+            serde_json::from_str::<HashSet<IpAddr>>(whitelist.as_str())
+                .map_err(|_| {
+                    BootstrapError::GeneralError(String::from(
+                        "Failed to parse bootstrap whitelist",
+                    ))
+                })?
+                .into_iter()
+                .map(normalize_ip)
+                .collect(),
+        )
+    } else {
+        None
+    };
+
+    let blacklist = if let Ok(blacklist) = std::fs::read_to_string(blacklist_path) {
+        Some(
+            serde_json::from_str::<HashSet<IpAddr>>(blacklist.as_str())
+                .map_err(|_| {
+                    BootstrapError::GeneralError(String::from(
+                        "Failed to parse bootstrap blacklist",
+                    ))
+                })?
+                .into_iter()
+                .map(normalize_ip)
+                .collect(),
+        )
+    } else {
+        None
+    };
+    Ok((whitelist, blacklist))
+}
+
 impl BootstrapServer {
     pub async fn run(mut self) -> Result<(), BootstrapError> {
         debug!("starting bootstrap server");
         massa_trace!("bootstrap.lib.run", {});
         let mut listener = self.establisher.get_listener(self.bind).await?;
         let mut bootstrap_sessions = FuturesUnordered::new();
-        // let cache_timeout = self.bootstrap_config.cache_duration.to_duration();
-        // let mut bootstrap_data: Option<(
-        //     BootstrapableGraph,
-        //     BootstrapPeers,
-        //     Arc<RwLock<FinalState>>,
-        // )> = None;
-        // let cache_timer = sleep(cache_timeout);
+        let cache_timeout = self.bootstrap_config.cache_duration.to_duration();
+        let (mut whitelist, mut blacklist) = reload_whitelist_blacklist(
+            &self.bootstrap_config.bootstrap_whitelist_path,
+            &self.bootstrap_config.bootstrap_blacklist_path,
+        )?;
+        let cache_timer = sleep(cache_timeout);
```

Comment:
I imagine that we would refresh the cache every X minutes, why not use [interval](https://docs.rs/tokio/latest/tokio/time/fn.interval.html)

Reply:
Done.